addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -1507,6 +1507,15 @@ def test_message_process_file_encoding_ascii(self):
         # This explains the multiple "�" in the attachment.
         self.assertIn("Chauss������e de Bruxelles", record.message_main_attachment_id.raw.decode())
 
+    def test_message_process_file_omitted_charset_with_bom(self):
+        """ Incoming email containing an xml attachment with omitted charset and starting with a byte order mark (BOM)
+        should not replace the BOM with replacement characters (otherwise, the parsing of the xml is broken).
+        """
+        record = self.format_and_process(test_mail_data.MAIL_MULTIPART_OMITTED_CHARSET_WITH_BOM, self.email_from, 'groups@test.com')
+        self.assertEqual(record.message_main_attachment_id.name, 'bis3.xml')
+        # EF BB BF is the BOM
+        self.assertEqual(b'\xef\xbb\xbf<Invoice>Test</Invoice>', record.message_main_attachment_id.raw)
```

Comment:
would be better to strip the bit-order-mask, also since "BOM" in Odoo is "Bill of Material" I would write "bit order mask (BOM)" in the comments instead of just BOM